### ISSUE TITLE
[Bug] pad_to_max_length silently disables the MLM objective in train.py

### ISSUE THREAD

**Issue by gss10282025** [Jiahao004/SimCSE-plus-plus#2]
### Summary

In `train.py`, `pad_to_max_length` also decides which collator is used:

```python
data_collator = default_data_collator if data_args.pad_to_max_length else OurDataCollatorWithPadding(tokenizer)
```

`OurDataCollatorWithPadding` is what builds `mlm_input_ids` and `mlm_labels` when `do_mlm=True`. So turning on fixed padding removes the MLM objective instead of only changing how tensors are padded.

The same line appears in `train_re_vicreg.py`; this report only tests `train.py`.

### Environment

Commit [51d2db55](https://github.com/Jiahao004/SimCSE-plus-plus/blob/51d2db55cf6009445558c9c9912ec887b6eb94a4/train.py), PyTorch 2.8.0+cu128, RTX 5090, FP64.

### Steps to reproduce

Run `train.py` with `do_mlm=True`, MLM weight `0.1`, `max_seq_length=8`, and four fixed two-view examples whose true length is 8. Compare `pad_to_max_length=False` and `True`.

Because every example already has length 8, both runs produce tensors of the same width. The only difference is the collator, so the comparison isolates the dispatch from any numerical effect of extra padding.

One update through the repository's Trainer, model, backward and AdamW: one-layer BERT (hidden size 32, vocabulary 128), dropout off, lr `0.001`, no weight decay or clipping, masking fixed by token identity so both runs mask the same positions.

| `pad_to_max_length=False` vs. `True`, `do_mlm=True` | Normalized L2 difference |
| --- | ---: |
| Full gradient, current code | `1.0001421324` |
| Full gradient, custom collator in both runs | `0` |

With the custom collator in both runs the loss, gradient, optimizer state and final state are identical. With `do_mlm=False` the two settings already agree. Reproduced on a second host.

### Expected behavior

The padding option should not decide whether an explicitly enabled objective exists.

### Proposed fix

Use `OurDataCollatorWithPadding` in both padding modes and let `pad_to_max_length` control only the sequence length. This is a deterministic FP64 check of the optional-MLM baseline entry; it says nothing about the paper's main recipe or STS scores.
